### PR TITLE
docs(homepage): revamp install UX & share how we build Lambda Layer

### PR DESCRIPTION
**Pull request by heitorlessa**
**Issue number:** #1942

### Changes

> Please provide a summary of what's being changed

This addresses @ericbn feedback on dependency resolution, how exactly we build the Lambda Layer, and make more visible how to inspect the contents of the Lambda Layer.

While I was at it, I've updated the new Powertools description, improved the UX on copying/pasting install commands, and made the `Install` section in the Local development slightly crispier too.

**Tasks**

* [x] Update Powertools project description
* [x] Improved UX for all install commands to be copied in one click (prevent error-prone typing)
* [x] Update Local Development section to include `[all]` dependency
* [x] Update Local Development section to explain what `[aws-sdk]` actually means for newcomers
* [x] Update Local Development section to explain dependency resolution as to why `[aws-sdk]` might end up with the latest AWS SDK version locally vs Lambda runtime
* [x] Update Lambda Layer section to include how we build the Layer
* [x] Move AWS CLI command to get Lambda Layer contents to be more visible - previously buried under Framework options

### User experience

> Please share what the user experience looks like before and after this change

image

image

image

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by heitorlessa** [comment]
hey @ericbn, I'd appreciate very much if you could have a look in the screenshots in the PR description to confirm whether that address your original ask.

Once again, **thank you** for helping us improve and make the documentation clearer for everyone!

Have a fantastic weekend.

File: docs/index.md [hunk 1]
```diff
@@ -24,41 +24,49 @@ A suite of utilities for AWS Lambda functions to ease adopting best practices su
...
+    * **Tracer** and **Parser**: **[`pip install "aws-lambda-powertools[tracer,parser]"`](#){: .copyMe}:clipboard:**
 
 ### Local development
 
-Powertools relies on the AWS SDK bundled in the Lambda runtime. This helps us achieve an optimal package size and initialization.
+!!! info "Using Powertools via Lambda Layer? Simply add [**`"aws-lambda-powertools[all]"`**](#){: .copyMe}:clipboard: as a development dependency."
+
+Powertools relies on the [AWS SDK bundled in the Lambda runtime](https://docs.aws.amazon.com/lambda/latest/dg/lambda-python.html){target="_blank"}. This helps us achieve an optimal package size and initialization.
+
+However, when developing locally, you might want to have IDE autocompletion, or run your tests suite locally without any emulation like [AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/install-sam-cli.html){target="_blank"}. For this to work, you need to install AWS SDK as a development dependency (not as a production dependency):
```
Comment (ericbn) [inline docs/index.md]: Can't we just simply say "However, when developing locally, you need to install AWS SDK as a development dependency (not as a production dependency)"?

I cannot think of any general scenario where developing locally without AWS SDK makes sense. Or maybe you were just enumerating scenarios where having the AWS SDK locally helps, which makes sense too.

File: docs/index.md [hunk 2]
```diff
@@ -24,41 +24,49 @@ A suite of utilities for AWS Lambda functions to ease adopting best practices su
...
-    * [Tracer](./core/tracer.md#install){target="_blank"} and [Parser](./utilities/parser.md#install){target="_blank"}: **`pip install "aws-lambda-powertools[tracer,parser]"`**
+    * **Tracer**: **[`pip install "aws-lambda-powertools[tracer]"`](#){: .copyMe}:clipboard:**
+    * **Validation**: **[`pip install "aws-lambda-powertools[validation]"`](#){: .copyMe}:clipboard:**
+    * **Parser**: **[`pip install "aws-lambda-powertools[parser]"`](#){: .copyMe}:clipboard:**
+    * **Tracer** and **Parser**: **[`pip install "aws-lambda-powertools[tracer,parser]"`](#){: .copyMe}:clipboard:**
 
 ### Local development
 
-Powertools relies on the AWS SDK bundled in the Lambda runtime. This helps us achieve an optimal package size and initialization.
+!!! info "Using Powertools via Lambda Layer? Simply add [**`"aws-lambda-powertools[all]"`**](#){: .copyMe}:clipboard: as a development dependency."
```
Comment (ericbn) [inline docs/index.md:45]: This is the sentence that I think summarizes my main doubt before. And having it stated as an instruction is perfect.

**Review by ericbn — COMMENTED** [review]
Olá @heitorlessa. Thanks for the updated documentation and the detailed images with all the changes and callouts! Amazing work, I love it! You have a fantastic weekend too.

File: docs/index.md (hunk 1 above)
Comment (heitorlessa) [reply docs/index.md]: The latter :) We have a few customers that aren't professional developers, so we try to raise awareness of why doing X could be helpful for the uninitiated.

That said, I'm gonna try another less verbose way to do that

Comment (heitorlessa) [reply docs/index.md]: Shorten the first part as you suggested, and added a long form under a banner that is folded by default to be less distracting. For anyone wanting to learn more, they can simply click to collapse to know the theory behind it.

image

**Comment by heitorlessa** [comment]
Made one tiny update to improve reading based on Eric's feedback!

image

**Comment by heitorlessa** [comment]
As @rubenfonseca is unavailable this week, and @leandrodamascena is off until end of March, I'm merging as both Eric (customer) and myself are happy with it.

**Pull request merged**